File: mix_player/ios/Classes/Models/BetterEventChannel.swift
//
//  BetterEventChannel.swift
//  AudioStreaming
//
//  Created by Dotsocket on 2/7/22.
//

import Foundation
import Flutter

class BetterEventChannel: NSObject,FlutterStreamHandler{
   
    
    var _eventChannel : FlutterEventChannel?
    var _eventSink : FlutterEventSink?
    
    init(name:String,message:FlutterBinaryMessenger){
        super.init()
        self._eventChannel = FlutterEventChannel(name: name, binaryMessenger: message)
        self._eventChannel?.setStreamHandler(self)
        self._eventSink = nil
       
    
    }
    
  
    func sendEvent(arguments:Any?){
        if(_eventSink != nil){
            _eventSink!(arguments)
        }
    }
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        _eventSink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        _eventSink = nil
        return nil
    }
    
   
}

File: mix_player/android/src/main/kotlin/com/example/mix_player/models/BetterEventChannel.kt
package com.example.mix_player.models
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class BetterEventChannel(name:String,message: FlutterPlugin.FlutterPluginBinding) :
    EventChannel.StreamHandler {
    var eventChannel: EventChannel = EventChannel(message.binaryMessenger, name)
    lateinit var eventSink:EventChannel.EventSink
    init {
        eventChannel.setStreamHandler(this)
    }

    fun sendEvent(arguments:Any?){
        eventSink.success(arguments)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events!!
        return
    }

    override fun onCancel(arguments: Any?) {
        eventSink.endOfStream()
        return
    }
}